core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/auth/AllOfGroupsRule.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.accesscontrol.blocks.rules.auth
+
+import tech.beshu.ror.accesscontrol.blocks.rules.Rule
+import tech.beshu.ror.accesscontrol.blocks.rules.Rule.*
+import tech.beshu.ror.accesscontrol.blocks.rules.auth.base.BaseGroupsRule
+import tech.beshu.ror.accesscontrol.blocks.rules.auth.base.BaseGroupsRule.{Creator, Settings}
+import tech.beshu.ror.accesscontrol.domain.*
+
+class AllOfGroupsRule(override val name: Rule.Name, override val settings: Settings[GroupsLogic.AllOf])
+                     (override implicit val userIdCaseSensitivity: CaseSensitivity)
+  extends BaseGroupsRule[GroupsLogic.AllOf](name, settings)
+
+object AllOfGroupsRule {
+  implicit val allOfTypeInfo: Creator[GroupsLogic.AllOf] = new AllOfGroupsRule(_, _)(_)
+
+  case object ExtendedSyntaxName extends RuleName[AllOfGroupsRule] {
+    override val name: Name = Rule.Name("all_of")
```

Comment:
I forgot about one more thing. Group name is used in audit, so it should be "stable identifier". We cannot change it depending on the configuration of the rule. I totally forgot about it.

So, I want to ask you (again, sorry) about changing it a little bit. For the rule name, let's pick the one that describes it clearly.  In this case, it would probably be "groups_all_of". Other names (all_of, groups_and, roles_and) won't be alternative names, but we will treat them as configuration keys only. The rule name should always be the same (and not passed with the constructor, but determined at the code level - as we do in the rest of ROR rules).

Reply:
I think it is now done as specified in this comment, please check.